src/app/_client-services/polkadot_api_service.ts

```
@@ -1146,4 +1150,338 @@ export class PolkadotApiService {
+		if (referendumInfo.isOngoing) {
+			const ongoing = referendumInfo.asOngoing;
+			const submittedAt = ongoing.submitted ? new BN(ongoing.submitted.toString()) : null;
+
+			// Calculate estimated unlock block for ongoing referendum
+			// This is when the referendum will end + lock period based on conviction
+			const convictionMultipliers = [0.1, 1, 2, 4, 8, 16, 32];
+			const lockBlocks = new BN(lockPeriod.toString()).muln(convictionMultipliers[conviction] || 0.1);
+
+			// Get the deciding period end from ongoing data
+			let estimatedEndBlock = currentBlock;
+			if (ongoing.deciding && ongoing.deciding.since) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)